Notifications v2: Improve styles of the user block

arthur791004 opened the pull request:
<!--
Link a related Github/Linear issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword. Consider using "part of" instead.
-->

Part of DOTDASH-296

## Proposed Changes

* Add locale to the app context
* Add external link to the note summary
* Improve styles of the user block

| Before | After |
| - | - |
|<img width="499" height="741" alt="image" src="https://github.com/user-attachments/assets/576fabec-9e42-454d-8f26-3a35eb0fc25a" />|<img width="492" height="736" alt="image" src="https://github.com/user-attachments/assets/40be07d2-476a-42f7-8b79-1087a20a79d8" />|


## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

*

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Go to /v2
* Open the notifications
* Switch to Likes
* Select any notification
* Ensure the styles look good

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you tested accessibility for your changes? Ensure the feature remains usable with various user agents (e.g., browsers), interfaces (e.g., keyboard navigation), and assistive technologies (e.g., screen readers) (PCYsg-S3g-p2).
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

arthur791004 committed:
Notifications v2: Add locale to the app context

arthur791004 committed:
Notifications v2: Improve styles of the user block

arthur791004 committed:
Fix styles priority

arthur791004 commented on apps/notifications/src/app/note/style.scss:
```diff
@@ -157,3 +165,40 @@
+	}
+
+	.wpnc__paragraph {
+		padding: 16px 0 0 !important;
```
Using `!important` as a workaround. Will remove it when we get rid of the v1 styles...

fushar reviewed (commented):
It seems there's a regression in the user avatar... see the weird tomato image in the reply below

|Before|After (this PR)|
|-|-|
|<img width="476" height="270" alt="Image" src="https://github.com/user-attachments/assets/2a1b8bab-dde8-4500-88c2-513229e63d74" />|<img width="479" height="274" alt="Image" src="https://github.com/user-attachments/assets/5390606e-b125-49d8-a9cd-86066de8ff6f" />

Also there's a weird external link below the reply arrow as seen above.

arthur791004 committed:
Fix overflow styles

arthur791004 committed:
Fix comment like styles

arthur791004 commented:
> This seems to be regression:

Fixed by https://github.com/Automattic/wp-calypso/pull/105417/commits/fa4821c6bc391e8b79e6b79b3fa3e31b52294e7b.

> It seems there's a regression in the user avatar... see the weird tomato image in the reply below

Fixed by https://github.com/Automattic/wp-calypso/pull/105417/commits/df899206795882a90431ba9d4a6d357e4a118360

> Also there's a weird external link below the reply arrow as seen above.

Fixed by https://github.com/Automattic/wp-calypso/pull/105409#issuecomment-3235786143.

arthur791004 committed:
Fix subscribe note

fushar reviewed (commented):
OK it's better but I think the image is somehow a bit off between the two "circles"? See the following two tomatoes; they are a bit different 🤔 they have different "padding"s

<img width="451" height="239" alt="Image" src="https://github.com/user-attachments/assets/b5a29ab8-e46d-46e4-8005-5a8ea1867eae" />

arthur791004 committed:
Fix the note icon in note summary is not in the center

arthur791004 commented:
Nice catch! It should be fixed by https://github.com/Automattic/wp-calypso/pull/105417/commits/f71872d67b67518139949950e67f49b97fb1dead.

fushar reviewed (approved):
Looks good now 👍
